contracts/main/stablecoin-core/StableSwapModuleWrapper.sol

```diff
@@ -138,15 +137,17 @@ contract StableSwapModuleWrapper is PausableUpgradeable, ReentrancyGuardUpgradea
                     _amount * WAD * tokenBalanceStableSwap18Decimals
                     /(stablecoinBalanceStableSwap18Decimals + tokenBalanceStableSwap18Decimals)
                     /WAD;
-                    
-        uint256 tokenAmountToWithdrawScaled = _convertDecimals(tokenAmountToWithdraw, 18, IToken(token).decimals());
         
-        depositTracker[msg.sender] -= _amount;
-        totalValueDeposited -= _amount;
 
+        require(stablecoinAmountToWithdraw + tokenAmountToWithdraw <= _amount, "withdrawTokens/total-amount-from-stableswap-exceeds-amount");
+        uint256 tokenAmountToWithdrawScaled = _convertDecimals(tokenAmountToWithdraw, 18, IToken(token).decimals());
+        
         _withdrawFromStableSwap(stablecoin, stablecoinAmountToWithdraw);
         _withdrawFromStableSwap(token, tokenAmountToWithdrawScaled);
 
+        depositTracker[msg.sender] -= _amount;
```

Comment:
here as well, I was thinking since deposit increases msg.sender by 2 * _amount, decreasing should happen in same manner,

Reply:
withdraw fn's _amount is in total <- Great Subik-ji comment